lib/src/dsp/editorhelper_neon.cpp

```diff
@@ -0,0 +1,381 @@
+  }
+}
+
+static void mirror_buffer_vertical_neon_uint64_t(uint64_t* src_buffer, uint64_t* dst_buffer,
+                                                 int src_w, int src_h, int src_stride,
+                                                 int dst_stride) {
+  uint64_t* src_row = src_buffer;
+  uint64_t* dst_row = dst_buffer + (src_h - 1) * dst_stride;
+
+  for (int i = 0; i < src_h; i++, src_row += src_stride, dst_row -= dst_stride) {
+    uint64_t* src_blk = src_row;
+    uint64_t* dst_blk = dst_row;
+    int j = 0;
+
+    for (; j + 2 <= src_w; src_blk += 2, dst_blk += 2, j += 2) {
+      uint64x2_t s0 = vld1q_u64(src_blk);
+      vst1q_u64(dst_blk, s0);
+    }
+
+    if (j < src_w) memcpy(dst_blk, src_blk, (src_w - j) * sizeof(uint64_t));
+  }
+}
+
+template <typename T>
+void mirror_buffer_neon(T* src_buffer, T* dst_buffer, int src_w, int src_h, int src_stride,
+                        int dst_stride, uhdr_mirror_direction_t direction) {
+  if (direction == UHDR_MIRROR_VERTICAL) {
+    if constexpr (sizeof(T) == 1) {
+      mirror_buffer_vertical_neon_uint8_t(src_buffer, dst_buffer, src_w, src_h, src_stride,
+                                          dst_stride);
+    } else if constexpr (sizeof(T) == 2) {
+      mirror_buffer_vertical_neon_uint16_t(src_buffer, dst_buffer, src_w, src_h, src_stride,
+                                           dst_stride);
+    } else if constexpr (sizeof(T) == 4) {
+      mirror_buffer_vertical_neon_uint32_t(src_buffer, dst_buffer, src_w, src_h, src_stride,
+                                           dst_stride);
+    } else if constexpr (sizeof(T) == 8) {
+      mirror_buffer_vertical_neon_uint64_t(src_buffer, dst_buffer, src_w, src_h, src_stride,
+                                           dst_stride);
+    }
+
+  } else if (direction == UHDR_MIRROR_HORIZONTAL) {
+    if constexpr (sizeof(T) == 1) {
+      mirror_buffer_horizontal_neon_uint8_t(src_buffer, dst_buffer, src_w, src_h, src_stride,
+                                            dst_stride);
+    } else if constexpr (sizeof(T) == 2) {
+      mirror_buffer_horizontal_neon_uint16_t(src_buffer, dst_buffer, src_w, src_h, src_stride,
+                                             dst_stride);
+    } else if constexpr (sizeof(T) == 4) {
+      mirror_buffer_horizontal_neon_uint32_t(src_buffer, dst_buffer, src_w, src_h, src_stride,
+                                             dst_stride);
+    } else if constexpr (sizeof(T) == 8) {
+      mirror_buffer_horizontal_neon_uint64_t(src_buffer, dst_buffer, src_w, src_h, src_stride,
+                                             dst_stride);
+    }
+  }
+}
+
+template <typename T>
+void rotate_buffer_clockwise_180_neon(T* src_buffer, T* dst_buffer, int src_w, int src_h,
+                                      int src_stride, int dst_stride) {
+  if constexpr (sizeof(T) == 1) {
+    mirror_buffer_horizontal_neon_uint8_t(src_buffer + (src_h - 1) * src_stride, dst_buffer, src_w,
```

Comment:
This may be incorrect. Rotate 180 degrees should be horizontal mirror then vertical mirror. Could you double check?